mockfactory/blinding/catalog.py

```diff
@@ -517,8 +523,16 @@ def png(self, data_positions, data_weights=None, randoms_positions=None, randoms
         shifts = recon.read_shifts(data_positions, position_type='pos', mpiroot=None, field='rsd')
         shifted_positions = data_positions - shifts
         recon.assign_data(shifted_positions, weights=data_weights, position_type='pos', mpiroot=None)
+
+        if recon.mpicomm.rank == 0: print(np.std(np.linalg.norm(shifts, axis=1)))
+
         if randoms_positions is not None:
             recon.assign_randoms(randoms_positions, weights=randoms_weights, position_type='pos', mpiroot=None)
+        if shotnoise_correction:
+            csum_data_weights = mpy.csize(data_positions) if data_weights is None else mpy.csum(data_weights)
```

Comment:
mpy.cshape(data_positions)[0] and mpy.cshape(randoms_positions)[0] might be better (though I agree the factor 3 cancels in the end)

Reply:
Ok, updated in the next commit